packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,581 @@
+  ///   @override
+  ///   State createState() => new SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final Map<int, Widget> children = <int, Widget>{};
+  ///
+  ///   int currentValue;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     children[0] = const Text('Child 1');
+  ///     children[1] = const Text('Child 2');
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Container(
+  ///       child:  new SegmentedControl<int>(
+  ///         children: children,
+  ///         onValueChanged: (int newValue) {
+  ///           setState(() {
+  ///             currentValue = newValue;
+  ///           });
+  ///         },
+  ///         groupValue: currentValue,
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  final ValueChanged<T> onValueChanged;
+
+  @override
+  _SegmentedControlState<T> createState() => _SegmentedControlState<T>();
+}
+
+class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
+  T _selectedValue;
+  T _pressedValue;
+  final List<Widget> _gestureChildren = <Widget>[];
+
+  @override
+  void initState() {
+    super.initState();
+    for (T currentKey in widget.children.keys) {
+      if (currentKey == widget.groupValue) {
+        _selectedValue = currentKey;
+      }
+    }
+  }
+
+  void _onTapDown(T currentKey) {
+    setState(() {
+      _pressedValue = currentKey;
+    });
+  }
+
+  void _onTapUp() {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTapCancel() {
+    setState(() {
+      _pressedValue = null;
+    });
+  }
+
+  void _onTap(T currentKey) {
+    setState(() {
+      if (_selectedValue != currentKey) {
+        widget.onValueChanged(currentKey);
+      }
+    });
+  }
+
+  @override
+  void didUpdateWidget(SegmentedControl<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    for (T currentKey in widget.children.keys) {
+      if (currentKey == widget.groupValue) {
+        _selectedValue = currentKey;
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _gestureChildren.clear();
+    for (T currentKey in widget.children.keys) {
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+            color: (_selectedValue == currentKey) ?
+            CupertinoColors.white : CupertinoColors.activeBlue,
+          );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedValue == currentKey) ?
+        CupertinoColors.white : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children[currentKey];
+      child = GestureDetector(
+        onTapDown: (TapDownDetails event) {
+          _onTapDown(currentKey);
+        },
+        onTapUp: (TapUpDetails event) {
+          _onTapUp();
+        },
+        onTapCancel: () {
+          _onTapCancel();
```

Comment:
just pass _onTapCancel directly

In fact for all of these (except onTap, which captures currentKey), just pass the methods directly, and adjust their signatures to match. That way you avoid the extraneous jmp.